### PR TITLE
Enhance local development setup instructions

### PR DESCRIPTION
**Pull request by that1matt**
- Add documentation on how to get a local development server up and running.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -72,6 +72,7 @@ services:
   postgres:
     image: postgis/postgis:13-3.1-alpine
     environment:
+      POSTGRES_USER: directus
```
Comment (that1matt) [inline docker-compose.yml:75]: This was needed to get the local API service to connect to the postgres docker container.

**Comment by paescuj** [comment]
Thanks for this pull request @that1matt! ❤️ 

We'd rather want to keep the `readme.md` file short & sweet.
This looks more like something for the [documentation](https://docs.directus.io).
Indeed we already have a similar manual there: https://docs.directus.io/contributing/running-locally.html
Maybe you could integrate your changes into this page? Before that, however, it would be useful if we could briefly discuss what you are missing in particular from this page / why you think the changes are necessary.

**Comment by that1matt** [comment]
@paescuj That sounds good. This wasn't easy to find, I will do some cleanup and push up some changes. I appreciate it.

File: readme.md [hunk 2]
```diff
@@ -44,6 +44,46 @@ Directus is a real-time API and App dashboard for managing SQL database content.
...
+3. Run `pnpm -r run build` to build all required package dependencies.
+4. Navigate to `/app` and run `pnpm dev`.
+5. Navigate to your local Directus URL, and login with the credentials you just created.
+
+### Changing Local Database with Docker
+
+- To change from a local `sqlite` database, you can use anyone of the supported databases in the root
+  [docker-compose.yml](./docker-compose.yml) by running `docker compose up -d postgres redis minio`.
+
+**Note:** If you are trying to run `postgres` on a M1/M2 MacBook (arm64), you will need to change the postgres image to
```
Comment (rijkvanzanten) [inline readme.md:83]: Is this required? I'm on a M1 Pro and the currently provided one seems to behave just fine 🤔

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Comment by phazonoverload** [comment]
> This wasn't easy to find

Just popping in as the current docs owner to say… I know. And I’m planning to do some new information architecture on the docs quite soon.

File: readme.md (hunk 2 above)
Comment (that1matt) [reply readme.md:83]: The original postgres image seems to have issues with my M1 MacBook. I found that this version significantly reduces the amount of memory required as it doesn't need to use Rosetta.

File: readme.md [hunk 3]
```diff
@@ -44,6 +44,46 @@ Directus is a real-time API and App dashboard for managing SQL database content.
 
 <br />
 
+## Local Development
```
[the remark this answers is not in the document]
Comment (paescuj) [reply readme.md:47]: You mean in the readme? There's already a link to the documentation 👍

File: readme.md (hunk 2 above)
Comment (that1matt) [reply readme.md:83]: I am happy to remove it if there is no value added.

Comment (paescuj) [reply readme.md:83]: How about newer versions of the official `postgis` image? I like to be a bit cautious when it comes to including unknown images in the documentation

File: readme.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (that1matt) [reply readme.md]: @paescuj I mean a reference to the getting a local development up and running. I was looking around the docs and didn't see anything that was easy to find. I was looking for different keywords when searching.

[the remark this answers is not in the document]
Comment (paescuj) [reply readme.md]: Oh okay, this one I'd like to defer to https://github.com/directus/directus/pull/18133#issuecomment-1503846830.

File: readme.md [hunk 4]
```diff
@@ -54,6 +54,9 @@ All security vulnerabilities should be reported in accordance with our
 Directus is a premium open-source ([GPLv3](./license)) project, made possible with support from our passionate core
 team, talented contributors, and amazing [GitHub Sponsors](https://github.com/sponsors/directus). Thank you all!
 
+See [Running Locally](https://docs.directus.io/contributing/running-locally.html) to get a local development version up
+and running.
+
```
Comment (paescuj) [inline readme.md:59]: Let's omit that in favor of https://github.com/directus/directus/pull/18133#issuecomment-1503846830

**Review by paescuj — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
Thanks @that1matt! 😄

**Comment by phazonoverload** [comment]
The docs now get deployed with new releases only, as a trial. It deploys with a merge to the `release` branch.

**Pull request merged**